lib/lever/client.rb

```diff
@@ -72,20 +108,41 @@ def post_resource(path, body)
       response.parsed_response
     end
 
+    def with_retries
+      return yield if using_with_retries
+
+      begin
+        # Eventually we want to have lower-level methods like #get_resource implement retries automatically
+        # So let's disallow nested `with_retries` blocks just in case we add it there but forget to remove it from
+        #   higher-level methods
+        self.using_with_retries = true
+
+        Retriable.retriable(on: [Lever::TooManyRequestsError, Lever::ServerError, Lever::ServiceUnavailableError]) do
+          yield
+        end
+      ensure
+        self.using_with_retries = false
+      end
+    end
+
     def get_resource(base_path, objekt, id = nil, options = {})
       path = id.nil? ? base_path : "#{base_path}/#{id}"
 
       add_query = options[:query]
       on_error = options[:on_error]
 
-      response = self.class.get("#{base_uri}#{path}", @options.merge({ query: add_query }))
+      response = self.class.get("#{base_uri}#{path}", @options.merge(query: add_query))
       if response.success?
+        parsed_response = response.parsed_response
+
+        yield parsed_response if block_given?
```

Comment:
This is a bit of a hack -- I needed this for: https://github.com/Grayscale-Labs/lever/blob/63a3fc9d423c345e3925d93cd8fdfc160729c495/lib/lever/resource_collection.rb#L71-L72

I think in the future, we should break this method into smaller methods or something b/c it's starting to get unwieldy. (#oneday)